libraries/AP_ADSB/sagetech-sdk/sgUtil.h

```diff
@@ -0,0 +1,325 @@
+/**
+ * @copyright Copyright (c) 2021 Sagetech, Inc. All rights reserved.
+ *
+ * @file   sgUtil.h
+ * @author jimb
+ *
+ * @date Feb 2, 2021
+ */
+
+#ifndef UTIL_H
+#define UTIL_H
+
+#include <stdint.h>
+
+#ifndef swap16
+#define swap16(data) \
+    ((((data) >> 8) & 0x00FF) | (((data) << 8) & 0xFF00))
+#endif
+
+#ifndef swap24
+#define swap24(data) \
+    (((data) >> 16) | ((data)&0x00FF00) | (((data) << 16) & 0xFF0000))
+#endif
+
+#ifndef swap32
+#define swap32(data) \
+    (((data) >> 24) | (((data)&0x00FF0000) >> 8) | (((data)&0x0000FF00) << 8) | ((data) << 24))
+#endif
+
+#ifndef swap64
+#define swap64(data) \
+   (swap32((data & 0x00000000ffffffffULL))) << 32 | swap32(data >> 32))
+#endif
+
+#ifndef PI
+#define PI 3.14159265359
+#endif
+
+#ifndef toRad
+#define toRad(deg) \
+    ((deg)*PI / 180.0)
+#endif
+
+#ifndef toDeg
+#define toDeg(rad) \
+    ((rad)*180 / PI)
+#endif
+
+#ifndef toMeter
+#define toMeter(feet) \
+    ((feet)*0.3048)
+#endif
+
+#ifndef toFeet
+#define toFeet(meter) \
+    ((meter)*3.2808)
+#endif
+
+/**
+ * Converts an array of bytes to a 16 bit integer.
+ *
+ * @param bytes The array of bytes to convert.
+ *
+ * @return The 16 bit integer.
+ */
+int16_t toInt16(const uint8_t bytes[]);
+
+/**
+ * Converts an array of bytes to a 32 bit integer.
+ *
+ * @param bytes The array of bytes to convert.
+ *
+ * @return The 32 bit integer.
+ */
+int32_t toInt32(const uint8_t bytes[]);
+
+/**
+ * Converts an array of bytes to a 16 unsigned bit integer.
+ *
+ * @param bytes The array of bytes to convert.
+ *
+ * @return The 16 bit integer.
+ */
+uint16_t toUint16(const uint8_t bytes[]);
+
+/**
+ * Converts an array of bytes to a 24 bit unsigned integer with leading 0s.
+ *
+ * @param bytes The array of bytes to convert.
+ *
+ * @return The 24 bit unsigned integer with leading 0s.
+ */
+uint32_t toUint24(const uint8_t bytes[]);
+
+/**
+ * Converts an array of bytes to a 32 bit unsigned integer.
+ *
+ * @param bytes The array of bytes to convert.
+ *
+ * @return The 32 bit unsigned integer.
+ */
+uint32_t toUint32(const uint8_t bytes[]);
+
+/**
+ * Converts an array of bytes to a distance.
+ *
+ * @param bytes The array of bytes to convert.
+ *
+ * @return The distance value.
+ */
+double toDist(const uint8_t *bytes);
+
+/**
+ * Converts an array of bytes to a 24 bit unsigned integer with leading 0's.
+ *
+ * @param bytes The array of bytes to convert.
+ *
+ * @return The 32 bit unsigned integer.
+ */
+uint32_t toIcao(const uint8_t bytes[]);
```

Comment:
function declaration ```toIcao()``` and it's file ```toIcao.c``` are not used.

In fact, there's 10 other fucntions in here that are declared in this h but have no source for and are unused.

Reply:
These are declarations to functions whose source has been removed to pare down our SDK for ardupilot. I can remove these but I had to add back some SDK source to meet the other suggestions you made (like adding decodeInstall to read the configuration on the MXS). I will remove these closer to the end once I'm sure there are no other items in the SDK we will need.